### PR TITLE
Ephemeral importer worker + async dirty

### PR DESCRIPTION
**Pull request by thalesmg**
Given that we observed `mria_rlog_replica` with huge queues waiting on `mria_replica_importer_worker:import_batch` during performance tests, there are 2 possible optimizations:

- Use an ephemeral importer worker, so that there's no garbage collection in it.
- Use `async_dirty` for everything when importing batches.

### CODE REVIEW

File: src/mria_rlog_replica.erl [hunk 1]
```diff
@@ -156,20 +155,28 @@ push_tlog_entry(sync, Shard, {Node, Pid}, Batch) ->
...
-    ok = mria_replica_importer_worker:import_batch(ImporterWorker, Transaction),
+    {Pid, Ref} = spawn_monitor(?MODULE, import_batch, [Transactions]),
+    receive
+        {'DOWN', Ref, process, Pid, normal} ->
+            ok;
+        {'DOWN', Ref, process, Pid, Reason} ->
+            exit({bad_import_result, Reason})
+    after
+        5_000 ->
+            exit(timeout)
```
Comment (k32) [inline src/mria_rlog_replica.erl:178]:
```suggestion
            exit(importer_timeout)
```

File: src/mria_rlog_replica.erl [hunk 2]
```diff
@@ -156,20 +155,28 @@ push_tlog_entry(sync, Shard, {Node, Pid}, Batch) ->
...
         #{ agent            => Agent
          , seqno            => SeqNo
-         , transaction      => Transaction
+         , transaction      => Transactions
          , tid              => _Tid
          }),
-    ok = mria_replica_importer_worker:import_batch(ImporterWorker, Transaction),
+    {Pid, Ref} = spawn_monitor(?MODULE, import_batch, [Transactions]),
+    receive
+        {'DOWN', Ref, process, Pid, normal} ->
```
Comment (k32) [inline src/mria_rlog_replica.erl:172]: :+1: for using ref trick. Maybe use a special exit status (like `imported`)?

File: src/mria_rlog_replica.erl [hunk 3]
```diff
@@ -156,20 +155,28 @@ push_tlog_entry(sync, Shard, {Node, Pid}, Batch) ->
...
     %% Normal flow, transactions are applied directly to the replica:
     ?tp(rlog_replica_import_trans,
         #{ agent            => Agent
          , seqno            => SeqNo
-         , transaction      => Transaction
+         , transaction      => Transactions
          , tid              => _Tid
          }),
-    ok = mria_replica_importer_worker:import_batch(ImporterWorker, Transaction),
+    {Pid, Ref} = spawn_monitor(?MODULE, import_batch, [Transactions]),
```
Comment (k32) [inline src/mria_rlog_replica.erl:170]: BTW, maybe dirty operations may be importerted in the main process. When async dirty is used, they shouldn't use any selective receives (I think), so they shouldn't be affected by long message queue in the main process.

File: src/mria_rlog_replica.erl [hunk 4]
```diff
@@ -440,3 +448,22 @@ post_connect(Shard, TableSpecs) ->
...
+
+import_batch_in_worker(Ops) ->
+    {Pid, Ref} = spawn_monitor(?MODULE, import_batch, [Ops]),
+    receive
+        {'DOWN', Ref, process, Pid, imported} ->
+            ok;
+        {'DOWN', Ref, process, Pid, Reason} ->
+            exit({bad_import_result, Reason})
+    after
+        5_000 ->
```
Comment (k32) [inline src/mria_rlog_replica.erl:463]: No need for a timeout here

Comment (thalesmg) [reply src/mria_rlog_replica.erl:463]: I was trying to preserve the previous behavior of the `gen_server:call` to the importer worker.  But now that we import dirty ops directly in the main process, that hasn't the same behavior as well.  I'll remove this.  :smile_cat:

File: src/mria_rlog_replica.erl [hunk 5]
```diff
@@ -440,3 +448,16 @@ post_connect(Shard, TableSpecs) ->
     Tables = [T || #?schema{mnesia_table = T} <- TableSpecs],
     mria_config:load_shard_config(Shard, Tables),
     ok = mria_schema:converge_replicant(Shard, TableSpecs).
+
+import_batch_in_worker(Ops) ->
+    {Pid, Ref} = spawn_monitor(?MODULE, import_batch, [Ops]),
```
Comment (k32) [inline src/mria_rlog_replica.erl:453]: Add a comment why we need a separate process here.

**Pull request closed without merging**